pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,412 @@
+			// Exact match will result in either pass or fail
+			if testCase.Name == id.testName {
+				if testCase.FailureOutput == nil {
+					return testPassed
+				} else {
+					return testFailed
+				}
+			}
+		}
+		return testSkipped
+	}
+	// Search next level
+	id.testSuites = id.testSuites[1:]
+	for _, childSuite := range testSuite.Children {
+		status := getTestStatus(id, childSuite)
+		if status == testPassed || status == testFailed {
+			return status
+		}
+	}
+	return testSkipped
+}
+
+func (r minimumRequiredPassesTestCaseChecker) addTestResultToDetails(currDetails *jobrunaggregatorlib.TestCaseDetails,
+	jobRun jobrunaggregatorapi.JobRunInfo, status testStatus) {
+	switch status {
+	case testPassed:
+		currDetails.Passes = append(
+			currDetails.Passes,
+			jobrunaggregatorlib.TestCasePass{
+				JobRunID:       jobRun.GetJobRunID(),
+				HumanURL:       jobRun.GetHumanURL(),
+				GCSArtifactURL: jobRun.GetGCSArtifactURL(),
+			})
+	case testFailed:
+		currDetails.Failures = append(
+			currDetails.Failures,
+			jobrunaggregatorlib.TestCaseFailure{
+				JobRunID:       jobRun.GetJobRunID(),
+				HumanURL:       jobRun.GetHumanURL(),
+				GCSArtifactURL: jobRun.GetGCSArtifactURL(),
+			})
+	default:
+		currDetails.Skips = append(
+			currDetails.Skips,
+			jobrunaggregatorlib.TestCaseSkip{
+				JobRunID:       jobRun.GetJobRunID(),
+				HumanURL:       jobRun.GetHumanURL(),
+				GCSArtifactURL: jobRun.GetGCSArtifactURL(),
+			})
+	}
+}
+
+// CheckTestCase returns a test case based on whether a test has passed certain criteria across job runs
+func (r minimumRequiredPassesTestCaseChecker) CheckTestCase(ctx context.Context, jobRuns []jobrunaggregatorapi.JobRunInfo) *junit.TestCase {
+	testName := fmt.Sprintf("test [%s/%s] should have required number of successful passes across jobRuns", strings.Join(r.id.testSuites, ":"), r.id.testName)
+	testCase := &junit.TestCase{
+		Name: testName,
+	}
+	start := time.Now()
+	successCount := 0
+	currDetails := &jobrunaggregatorlib.TestCaseDetails{
+		Name:          r.id.testName,
+		TestSuiteName: strings.Join(r.id.testSuites, ":"),
+	}
+	for i := range jobRuns {
+		jobRun := jobRuns[i]
+		status := testSkipped
+
+		testSuites, err := jobRun.GetCombinedJUnitTestSuites(ctx)
+		if err != nil {
+			continue
+		}
+
+		// Now go through test suites check test result
+		for _, testSuite := range testSuites.Suites {
+			status = getTestStatus(r.id, testSuite)
+			if status == testFailed {
+				break
+			} else if status == testPassed {
+				successCount++
+				break
+			}
+		}
+		r.addTestResultToDetails(currDetails, jobRun, status)
+	}
+	currDetails.Summary = fmt.Sprintf("Total job runs: %d, passes: %d, failures: %d, skips %d", len(jobRuns), len(currDetails.Passes), len(currDetails.Failures), len(currDetails.Skips))
+	detailsYaml, err := yaml.Marshal(currDetails)
+	if err != nil {
+		return nil
+	}
+	testCase.Duration = time.Since(start).Seconds()
+	testCase.SystemOut = string(detailsYaml)
```

Comment:
I actually thought setting SystemOut = automatic failure, but may have been mistaken or that was in another area.

Reply:
That is FailureOutput I think.